### PR TITLE
Add support for Vlive.tv VOD's

### PR DESCRIPTION
**Pull request by Noxime**
Preliminary support for https://vlive.tv. Due to Rsgets design I don't think streams are currently possible to implement. Hopefully for rsget 2.0?

### CODE REVIEW

File: rsget_lib/src/plugins/vlive.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            .ok_or(StreamError::Rsget(RsgetError::new("No capture found")))?[2]
+            .to_string();
+        let chan = vid_chan_re
+            .captures(&page)
+            .ok_or(StreamError::Rsget(RsgetError::new("No capture found")))?[1]
+            .to_string();
+
+        let page_req = dc.make_request(&format!("https://global.apis.naver.com/rmcnmv/rmcnmv/vod_play_videoInfo.json?key={}&videoId={}", key, id), None)?;
+
+        // all these structs are quite excessive for what we actually need but i want to be ready for the "Quality Update"
```
Comment (Erk-) [inline rsget_lib/src/plugins/vlive.rs]: Is there a reason you don't define the below struts at the top level?

File: rsget_lib/src/plugins/vlive.rs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+use crate::utils::error::StreamError;
+
+#[derive(Debug, Clone)]
+pub struct Vlive {
+    client: DownloadClient,
+    url: String,
+    title: String,
+    author: String,
+    video_url: Option<String>,
+    // TODO FOR ERK: This field is currently unused. This is due to Rsgets design being too focused on making plugin
```
Comment (Erk-) [inline rsget_lib/src/plugins/vlive.rs]: Maybe this should be moved into a seperate issue about redesign.

Comment (Noxime) [reply rsget_lib/src/plugins/vlive.rs]: Yes, I do admit a long rant in comments is perhaps not the best form to communicate issues.

File: rsget_lib/src/plugins/vlive.rs (hunk 1 above)
Comment (Noxime) [reply rsget_lib/src/plugins/vlive.rs]: Those structs exist only for the json deserialization and will not be used from any other function in the module. I personally like to keep definitions is as tight scopes as possible as to not litter the module with one-off types

Comment (Erk-) [reply rsget_lib/src/plugins/vlive.rs]: That is fine by me then 👍

**Pull request merged**